src/Utility/Security.php

```diff
@@ -388,4 +388,33 @@ public static function salt($salt = null)
 
         return static::$_salt = (string)$salt;
     }
+
+    /**
+     * Creates a secure random sensitive case token.
+     *
+     * @param int $length String length. Default 12.
+     * @return string
+     */
+    public static function randomToken($length = 12)
+    {
+        $random = base64_encode(static::randomBytes($length * 4));
+        $clean = preg_replace('/[^A-Za-z0-9]/', '', $random);
+
+        return substr($clean, random_int(1, $length * 2), $length);
+    }
+
+    /**
+     * Generate a 12 chars short hash (like git signature).
+     *
+     * @param  mixed $input Data to hash. Can be string or array.
+     * @return string
+     */
+    public static function shortHash($input)
```

Comment:
This is kind of "false sense of security" territory and IMHO shouldn't be in the core, shortening a hash has rather limited use cases, and it can easily turn into a security problem when not handled with the necessary care!

Git's short hashes are pretty much just aliases that may be used for convenience, internally actual operations will use the full hash. Also they're not just shortened, but shortened with respect to the existing commit hashes, where the specified length is a minimum value that is being increased when required in order to avoid collisions.

Reply:
I agree a short hash cannot be used in a security context as it won't have enough entropy to keep it unique or safe from timing/dictionary attacks.